x-pack/test/rule_registry/security_and_spaces/tests/basic/get_alert_by_id.ts

```diff
@@ -0,0 +1,219 @@
+      (indexName) => indexName === APM_ALERT_INDEX
+    );
+    expect(observabilityIndex).to.eql(APM_ALERT_INDEX); // assert this here so we can use constants in the dynamically-defined test cases below
+  };
+
+  const getSecuritySolutionIndexName = async (user: User) => {
+    const {
+      body: indexNames,
+    }: { body: { index_name: string[] | undefined } } = await supertestWithoutAuth
+      .get(`${getSpaceUrlPrefix(SPACE1)}${ALERTS_INDEX_URL}`)
+      .auth(user.username, user.password)
+      .set('kbn-xsrf', 'true')
+      .expect(200);
+    const securitySolution = indexNames?.index_name?.find(
+      (indexName) => indexName === SECURITY_SOLUTION_ALERT_INDEX
+    );
+    expect(securitySolution).to.eql(SECURITY_SOLUTION_ALERT_INDEX); // assert this here so we can use constants in the dynamically-defined test cases below
+  };
+
+  describe('Alerts - GET - RBAC - spaces', () => {
+    before(async () => {
+      await getSecuritySolutionIndexName(superUser);
+      await getAPMIndexName(superUser);
+
+      await esArchiver.load('x-pack/test/functional/es_archives/rule_registry/alerts');
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/rule_registry/alerts');
+    });
+
+    it('should return a 404 when superuser accesses not-existent alert', async () => {
+      await supertestWithoutAuth
+        .get(`${getSpaceUrlPrefix()}${TEST_URL}?id=myfakeid&index=${APM_ALERT_INDEX}`)
+        .auth(superUser.username, superUser.password)
+        .set('kbn-xsrf', 'true')
+        .expect(404);
+    });
+
+    it('should return a 404 when superuser accesses not-existent alerts as data index', async () => {
+      await supertestWithoutAuth
+        .get(`${getSpaceUrlPrefix()}${TEST_URL}?id=${APM_ALERT_ID}&index=myfakeindex`)
+        .auth(superUser.username, superUser.password)
+        .set('kbn-xsrf', 'true')
+        .expect(404);
+    });
+
+    function addTests({ space, authorizedUsers, unauthorizedUsers, alertId, index }: TestCase) {
+      authorizedUsers.forEach(({ username, password }) => {
+        it(`${username} should be able to access alert ${alertId} in ${space}/${index}`, async () => {
+          await supertestWithoutAuth
+            .get(`${getSpaceUrlPrefix(space)}${TEST_URL}?id=${alertId}&index=${index}`)
+            .auth(username, password)
+            .set('kbn-xsrf', 'true')
+            .expect(200);
+        });
+
+        // // NOTE: this test case actually results in a 500 error, not 404
+        // it(`${username} should fail to access a non-existent alert in ${space}/${index}`, async () => {
+        //   const fakeAlertId = 'some-alert-id-that-doesnt-exist';
+        //   await supertestWithoutAuth
+        //     .get(`${getSpaceUrlPrefix(space)}${TEST_URL}?id=${fakeAlertId}&index=${index}`)
+        //     .auth(username, password)
+        //     .set('kbn-xsrf', 'true')
+        //     .expect(404);
+        // });
```

Comment:
If you remove the superuser-specific tests above, you can uncomment this and add an additional case for missing index that will be exercised for all users:

```suggestion
        it(`${username} should fail to access a non-existent alert in ${space}/${index}`, async () => {
          const fakeAlertId = 'some-alert-id-that-doesnt-exist';
          await supertestWithoutAuth
            .get(`${getSpaceUrlPrefix(space)}${TEST_URL}?id=${fakeAlertId}&index=${index}`)
            .auth(username, password)
            .set('kbn-xsrf', 'true')
            .expect(404);
        });

        it(`${username} should fail to access alert ${alertId} in a non-existent index in ${space}`, async () => {
          const fakeIndexId = 'some-index-that-doesnt-exist';
          await supertestWithoutAuth
            .get(`${getSpaceUrlPrefix(space)}${TEST_URL}?id=${alertId}&index=${fakeIndexId}`)
            .auth(username, password)
            .set('kbn-xsrf', 'true')
            .expect(404);
        });
```